In [1]:
# 01_data_acquisition.ipynb
# Purpose: Load the UCI Concrete Compressive Strength dataset,
# inspect the raw data, clean column names, and save a processed CSV.

import pandas as pd
import numpy as np
from pathlib import Path

In [2]:
# Define file paths
data_dir = Path("../data")
raw_file = data_dir / "concrete+compressive+strength" / "Concrete_Data.xls"

# Load the raw Excel dataset
df = pd.read_excel(raw_file)

df.head()

,Cement (component 1)(kg in a m^3 mixture),Blast Furnace Slag (component 2)(kg in a m^3 mixture),Fly Ash (component 3)(kg in a m^3 mixture),Water (component 4)(kg in a m^3 mixture),Superplasticizer (component 5)(kg in a m^3 mixture),Coarse Aggregate (component 6)(kg in a m^3 mixture),Fine Aggregate (component 7)(kg in a m^3 mixture),Age (day),"Concrete compressive strength(MPa, megapascals)"
0,540.0,0.0,0.0,162.0,2.5,1040.0,676.0,28,79.986111
1,540.0,0.0,0.0,162.0,2.5,1055.0,676.0,28,61.887366
2,332.5,142.5,0.0,228.0,0.0,932.0,594.0,270,40.269535
3,332.5,142.5,0.0,228.0,0.0,932.0,594.0,365,41.052780
4,198.6,132.4,0.0,192.0,0.0,978.4,825.5,360,44.296075


In [3]:
print("Dataset shape:", df.shape)

print("\nMissing values:")
print(df.isnull().sum())

Dataset shape: (1030, 9)

Missing values:
Cement (component 1)(kg in a m^3 mixture)                0
Blast Furnace Slag (component 2)(kg in a m^3 mixture)    0
Fly Ash (component 3)(kg in a m^3 mixture)               0
Water  (component 4)(kg in a m^3 mixture)                0
Superplasticizer (component 5)(kg in a m^3 mixture)      0
Coarse Aggregate  (component 6)(kg in a m^3 mixture)     0
Fine Aggregate (component 7)(kg in a m^3 mixture)        0
Age (day)                                                0
Concrete compressive strength(MPa, megapascals)          0
dtype: int64


In [4]:
df.columns = [
    "cement",
    "slag",
    "fly_ash",
    "water",
    "superplasticizer",
    "coarse_aggregate",
    "fine_aggregate",
    "age",
    "compressive_strength"
]

df.head()

,cement,slag,fly_ash,water,superplasticizer,coarse_aggregate,fine_aggregate,age,compressive_strength
0,540.0,0.0,0.0,162.0,2.5,1040.0,676.0,28,79.986111
1,540.0,0.0,0.0,162.0,2.5,1055.0,676.0,28,61.887366
2,332.5,142.5,0.0,228.0,0.0,932.0,594.0,270,40.269535
3,332.5,142.5,0.0,228.0,0.0,932.0,594.0,365,41.052780
4,198.6,132.4,0.0,192.0,0.0,978.4,825.5,360,44.296075


In [5]:
df["w_c_ratio"] = df["water"] / df["cement"]

df[["water", "cement", "w_c_ratio"]].head()

,water,cement,w_c_ratio
0,162.0,540.0,0.300000
1,162.0,540.0,0.300000
2,228.0,332.5,0.685714
3,228.0,332.5,0.685714
4,192.0,198.6,0.966767


In [6]:
import numpy as np

df["log_age"] = np.log(df["age"])

df[["age", "log_age"]].head()

,age,log_age
0,28,3.332205
1,28,3.332205
2,270,5.598422
3,365,5.899897
4,360,5.886104


In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1030 entries, 0 to 1029
Data columns (total 11 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   cement                1030 non-null   float64
 1   slag                  1030 non-null   float64
 2   fly_ash               1030 non-null   float64
 3   water                 1030 non-null   float64
 4   superplasticizer      1030 non-null   float64
 5   coarse_aggregate      1030 non-null   float64
 6   fine_aggregate        1030 non-null   float64
 7   age                   1030 non-null   int64  
 8   compressive_strength  1030 non-null   float64
 9   w_c_ratio             1030 non-null   float64
 10  log_age               1030 non-null   float64
dtypes: float64(10), int64(1)
memory usage: 88.6 KB


In [8]:
processed_file = "../data/concrete_processed.csv"

df.to_csv(processed_file, index=False)

print("Saved:", processed_file)

Saved: ../data/concrete_processed.csv


In [9]:
df["compressive_strength"].describe()

count    1030.000000
mean       35.817836
std        16.705679
min         2.331808
25%        23.707115
50%        34.442774
75%        46.136287
max        82.599225
Name: compressive_strength, dtype: float64

## Summary

*This notebook loaded the UCI Concrete Compressive Strength dataset, inspected the raw data, confirmed that the dataset contains 1,030 rows and no missing values, renamed the columns for easier analysis, and created two engineered features based on instructor feedback: water-cement ratio (`w_c_ratio`) and logarithmic curing age (`log_age`). The processed dataset was saved as `data/concrete_processed.csv` for use in the EDA and modeling notebooks.*